### ISSUE TITLE
[Request] Add OpenTofu support for node auto-deployment

### ISSUE THREAD

**Issue by FPVogel** [pterodactyl/panel#5086]
### Is there an existing feature request for this?

- [X] I have searched the existing issues before opening this feature request.

### Describe the feature you would like to see.

We are currently running a multi-cloud architecture (primarily so that if shit hits the fan, it's about 45 minutes to migrate and be back up and running again, as well as offering more locations.) Therefore it would be really cool to see Terraform/OpenTofu support added to Wings deployment, which can dynamically deploy instances based on how they are needed. The problem with offering lots of locations imho is having a few locations that constantly need new nodes, because they are being excessively booked, while other locations each have a handful of servers just idling away.

### Describe the solution you'd like.

Therefore what I (and probably others) would like to see is support for OpenTofu, wherein you set up the corresponding cloud providers and then as nodes are being booked, new nodes are being created per location instead of booking particular nodes.
Basically going by the following flow:
```Customer books service in region x -> Ptero: Is there a node in region x and is there enough capacity to host the booked service? -> If no, break normal wings deployment flow -> notify customer that this might take two minutes -> deploy wings node using OpenTofu -> continue as normal ```
This would alleviate quite a bit of work setting up new nodes manually, especially in low-traffic regions, as well as saving server costs and sparing unnecessarily booked resources at cloud providers.

### Additional context to this request.

I know that this would probably change a significant portion of the deployment flow, but it would be hugely helpful for at least us (and almost certainly others)

**Comment by matthewpi** [comment]
If someone else wants to build support for something like this externally, I'd encourage it.  If there is a need for modifications to the Pterodactyl API to be able to facilitate this external feature I'd be open to implementing it.

But I don't have the time or want to build something like this natively into Pterodactyl, especially since there is no reason for it to be.
